File: ros2_ws/src/ros2_app_py/ros2_app_py/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int64

class NodePublisherNode(Node):
	def __init__(self):
		super().__init__("number_publisher")
		self.declare_parameter("number_to_publish", 2)
		self.declare_parameter("number_publish_frecuency", 1)

		self.number_ = self.get_parameter("number_to_publish").value
		self.publish_frecuency_ = self.get_parameter("number_publish_frecuency").value

		self.pub_ = self.create_publisher(Int64, "number", 10)

		self.number_timer_ = self.create_timer(1/self.publish_frecuency_, self.publish_number)
		self.get_logger().info("Number publisher has been started!")

	def print_number(self):
		self.get_logger().info(str(self.number_))

	def publish_number(self):
		msg = Int64()
		msg.data = self.number_
		self.pub_.publish(msg)

def main(args=None):
	rclpy.init(args=args)
	node = NodePublisherNode()
	rclpy.spin(node)
	rclpy.shutdown()

if __name__ == '__main__':
	main()

File: ros2_ws/src/ros2_app_cpp/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/int64.hpp"

class NumberPublisherNode : public rclcpp::Node
{
public:
	NumberPublisherNode(): Node("number_publisher")
	{
		this->declare_parameter("number_to_publish",2);
		this->declare_parameter("number_publish_frecuency",1.0);

		number_ = this->get_parameter("number_to_publish").as_int();
		double publish_frecuency = this->get_parameter("number_publish_frecuency").as_double();

		number_publisher_ = this->create_publisher<std_msgs::msg::Int64>("number", 10);

		number_timer_ = this->create_wall_timer(std::chrono::milliseconds((int)(1000.0/publish_frecuency)), 
																						std::bind(&NumberPublisherNode::publishNumber, this));
		RCLCPP_INFO(this->get_logger(), "Number publisher has been started");
	}


private:
	void publishNumber()
	{
		auto msg = std_msgs::msg::Int64();
		msg.data = number_;
		number_publisher_->publish(msg);
	}

	int number_;
	rclcpp::Publisher<std_msgs::msg::Int64>::SharedPtr number_publisher_;
	rclcpp::TimerBase::SharedPtr number_timer_;
};

int main(int argc, char **argv)
{
	rclcpp::init(argc, argv);
	auto node = std::make_shared<NumberPublisherNode>();

	rclcpp::spin(node);
	rclcpp::shutdown();
	return 0;
}